From 26ac873fe115d1081f9b95e510c8845e5c5fc08b Mon Sep 17 00:00:00 2001
From: Greg Moffatt <greg.moffatt@windriver.com>
Date: Tue, 21 Dec 2010 09:15:02 -0500
Subject: [PATCH 05/11] kernel: use SLAB allocator with NUMA

When NUMA is configured for a CGL kernel on an intel 5500 target,
the kernel panics when init is started.  There is a conflict between
NUMA, GRSEC/PAX, and the SLUB allocator.

To address this the SLAB allocator should be configured instead.  To
achieve this the Kconfig for the allocator is changed such that if
NUMA is configured by a user the SLAB allocator is selected, rather
than the SLUB allocator.

Signed-off-by: Greg Moffatt <greg.moffatt@windriver.com>
Signed-off-by: Joe MacDonald <joe.macdonald@windriver.com>
---
 init/Kconfig |    1 +
 1 file changed, 1 insertion(+)

diff --git a/init/Kconfig b/init/Kconfig
index 336a46c..790b5f0 100644
--- a/init/Kconfig
+++ b/init/Kconfig
@@ -1340,6 +1340,7 @@ config COMPAT_BRK
 
 choice
 	prompt "Choose SLAB allocator"
+	default SLAB if (NUMA && GRKERNSEC)
 	default SLUB
 	help
 	   This option allows to select a slab allocator.
-- 
1.7.9.7

